crates/fee/src/lib.rs

```diff
@@ -199,73 +192,63 @@ decl_module! {
         // Initialize events
         fn deposit_event() = default;
 
-        /// Upgrade the runtime depending on the current `StorageVersion`.
-        fn on_runtime_upgrade() -> Weight {
-            if Self::storage_version() == Version::V0 {
-                let next_account_id = Self::fee_pool_account_id();
-                let prev_account_id = <FeePoolAccountId<T>>::take();
-
-                // transfer collateral
-                let amount_backing = ext::collateral::get_free_balance::<T>(&prev_account_id);
-                ext::collateral::transfer::<T>(prev_account_id.clone(), next_account_id.clone(), amount_backing).expect("failed to transfer collateral");
-
-                // tranfer tokens
-                let amount_btc = ext::treasury::get_free_balance::<T>(prev_account_id.clone());
-                ext::treasury::transfer::<T>(prev_account_id, next_account_id, amount_btc).expect("failed to transfer tokens");
-
-                StorageVersion::put(Version::V1);
-            }
-
-            0
+        /// Withdraw all vault backing rewards.
+        ///
+        /// # Arguments
+        ///
+        /// * `origin` - signing account
+        #[weight = <T as Config>::WeightInfo::withdraw_vault_rewards()]
+        #[transactional]
+        fn withdraw_vault_backing_rewards(origin) -> DispatchResult
+        {
+            ext::security::ensure_parachain_status_not_shutdown::<T>()?;
+            let signer = ensure_signed(origin)?;
+            Self::withdraw_backing::<T::BackingVaultRewards>(&signer)?;
+            Ok(())
         }
 
-        fn on_initialize(n: T::BlockNumber) -> Weight {
-            if let Err(e) = Self::begin_block(n) {
-                sp_runtime::print(e);
-            }
-            // TODO: calculate weight
-            0
+        /// Withdraw all vault issuing rewards.
+        ///
+        /// # Arguments
+        ///
+        /// * `origin` - signing account
+        #[weight = <T as Config>::WeightInfo::withdraw_vault_rewards()]
+        #[transactional]
+        fn withdraw_vault_issuing_rewards(origin) -> DispatchResult
+        {
+            ext::security::ensure_parachain_status_not_shutdown::<T>()?;
+            let signer = ensure_signed(origin)?;
+            Self::withdraw_issuing::<T::IssuingVaultRewards>(&signer)?;
+            Ok(())
         }
 
-        /// Allows token reward withdrawal if balance is sufficient.
+        /// Withdraw all relayer backing rewards.
         ///
         /// # Arguments
         ///
         /// * `origin` - signing account
-        /// * `amount` - amount of Issuing
-        #[weight = <T as Config>::WeightInfo::withdraw_issuing()]
+        #[weight = <T as Config>::WeightInfo::withdraw_relayer_rewards()]
         #[transactional]
-        fn withdraw_issuing(origin, #[compact] amount: Issuing<T>) -> DispatchResult
+        fn withdraw_relayer_backing_rewards(origin) -> DispatchResult
```

Comment:
Why not simplify this from four to one function: `withdraw_reward(origin, AccountType, CurrencyType)`? 

Do nominators withdraw via the same functions?

Reply:
I think we will have to add separate methods on the `vault-registry` and `nomination` pallets to do the fee withdrawals.

Reply:
We can add a `CurrencyType` but I'm not sure where to define this. If we do end up adopting something like `tokens` from `orml` then would make most sense to define it there. Using `tokens` instead of multiple `pallet_balances` instances will also reduce some complexity in other areas tbh.

Reply:
Ok, then probably makes sense to switch over to `tokens` in a later PR. Happy to keep the four functions as well, just thought the API might look cleaner if we don't define parameters via the function names.